### PR TITLE
Support PNG avatars with transparent backgrounds and fix EXIF rotation

### PR DESCRIPTION
**Pull request by Zeokat**
With this PR i try to solve the issues:

+ [1161](https://github.com/flarum/core/issues/1161): Upload profile image error - php exif extension
+ [1163](https://github.com/flarum/core/issues/1163): Problems uploading PNG images as avatars with transparent backgrounds

I decided to solve both issues with one PR, because both are related.

### CODE REVIEW

**Comment by Luceos** [comment]
Thanks for the cross-reference. 👍

File: src/Core/Command/UploadAvatarHandler.php [hunk 1]
```diff
@@ -101,7 +101,7 @@ public function handle(UploadAvatar $command)
             $manager = new ImageManager;
 
             // Explicitly tell Intervention to encode the image as JSON (instead of having to guess from the extension)
-            $encodedImage = $manager->make($tmpFile)->orientate()->fit(100, 100)->encode('jpg', 100);
+            $encodedImage = $manager->make($tmpFile)->fit(100, 100)->encode('png', 100);
```
Comment (franzliedke) [inline src/Core/Command/UploadAvatarHandler.php]: Can you fix the comment in the line above? That says JSON and probably meant JPEG. Plus now it's PNG ;)

**Comment by franzliedke** [comment]
Also, please check for EXIF being available and call `orientate()` if so, as @kulga suggested in https://github.com/flarum/core/issues/1161#issue-222779188.

**Comment by Zeokat** [comment]
I can't understand clearly your last message @franzliedke . I already checked usage of `orientate()` into a server with a PHP-EXIF module enabled and no errors throwed on screen and `/storage/logs/` remains empty.

**Comment by datitisev** [comment]
@Zeokat Franz is saying that you deleted the `orientate()` method call, and that you should add it back for it to execute only if the server has EXIF available.

screen shot 2017-04-29 at 10 00 19 am

**Comment by Zeokat** [comment]
Oh i see @datitisev , but honestly... code that stuff isn't so trivial becasue we not only need to check if `EXIF` is enanled. On windows servers for example, it also requires `mbstring`. So it's not only check if EXIF is enabled, is check Operating System and then check modules required. Two checks for something useless like the usage of `orientate`... has no sense.

As i said into the issue opened by Kuga https://github.com/flarum/core/issues/1161 , has no sense use `orientate`, we need avatars, we don't need read EXIF to rotate the image.

**Comment by franzliedke** [comment]
@Zeokat mbstring is already part of our [System Requirements](http://flarum.org/docs/installation/#system-requirements).

It's not useless, that we can offer (because we already did) when the server supports it. Would be nice if you could add it in.

**Comment by Zeokat** [comment]
Ok @franzliedke i will try to implement that.

**Comment by franzliedke** [comment]
Thanks for your patience!

**Pull request merged**